### PR TITLE
Sherwani/add api key

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [x] I have tested the new feature or updates in both light and dark themes.
- [x] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [x] I have tested my changes on mobile screen sizes.
- [x] I have tested my changes on tablet screen sizes.
- [x] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [x] Explain the task/bug you are solving
- [x] Show the changes in the UI
- [x] Show the changes in the codebase
- [x] Show the changes you tested to make sure it works
- [x] Show the changes in various screen sizes
- [x] Show the changes in both light and dark themes
video:https://www.loom.com/share/d8477949c1454869b1a34aaf8e7909cb?sid=752fb9da-c85c-44d5-8369-28ef395106b7

### CODE REVIEW

File: academy-ui/src/app/api/spaces/add-new-api-key/route.ts
Comment (RobinNagpal) [inline academy-ui/src/app/api/spaces/add-new-api-key/route.ts:1]: We can update the path to `/api/actions/spaces/generate-api-key/route.ts`

File: academy-ui/src/components/spaces/Edit/ApiKey/upsertSpaceApiKeyModal.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      .writeText(text)
+      .then(() => {
+        showNotification({ type: 'success', message: 'API key copied to Clipboard better save it somewhere' });
+      })
+      .catch((err) => {
+        showNotification({ type: 'error', message: 'Not able to copy to Clipboard. Try Again!' });
+      });
+  };
+  return (
+    <FullPageModal
```
Comment (RobinNagpal) [inline academy-ui/src/components/spaces/Edit/ApiKey/upsertSpaceApiKeyModal.tsx:38]: Try using SingleSectionModal if that looks better.

Arrange the buttons as discussed, and use the copy icon

Comment (RobinNagpal) [reply academy-ui/src/components/spaces/Edit/ApiKey/upsertSpaceApiKeyModal.tsx:38]: Use the highlighted icon. We already use hero icons at multiple places

image

File: academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+export default function SpaceApiKeyDetails(props: SpaceApiKeyDetailsProps) {
+  const [showApiKeyModal, setshowApiKeyModal] = useState(false);
+  const [space, setSpace] = useState(props.space);
+
+  return (
+    <>
+      <DetailsSection className={`${props.className} shadow`}>
```
Comment (RobinNagpal) [inline academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx:39]: Make it look like the existing design. And you can show a button instead of elipsis

image


We can also add  a 'size' prop to button, which is an enum. ButtonSize {Small, Medium, Large}, In this case we can pass Small, which should make the button smaller

image

File: academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        <DetailsHeader header={'API Key Details'} className="grow-1 w-full" />
+        <div className="flex flex-col w-full">
+          {getSpaceDetailsFields(space).map((field) => (
```
Comment (RobinNagpal) [inline academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx:42]: If no keys are there we can definitely show some message, that you can add an API key.

File: academy-ui/src/app/api/spaces/add-new-api-key/route.ts
Comment (HussainKhanSherwani) [reply academy-ui/src/app/api/spaces/add-new-api-key/route.ts:1]: done

File: academy-ui/src/components/spaces/Edit/ApiKey/upsertSpaceApiKeyModal.tsx (hunk 1 above)
Comment (HussainKhanSherwani) [reply academy-ui/src/components/spaces/Edit/ApiKey/upsertSpaceApiKeyModal.tsx]: done

File: academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx (hunk 2 above)
Comment (HussainKhanSherwani) [reply academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx:37]: done

File: academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx (hunk 3 above)
Comment (HussainKhanSherwani) [reply academy-ui/src/components/spaces/View/SpaceApiKeysDetails.tsx]: done
